plugin/pkg/admission/disableservicelinks/admission_test.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package disableservicelinks
+
+import (
+	"context"
+	"testing"
+
+	"k8s.io/apiserver/pkg/admission"
+	admissiontesting "k8s.io/apiserver/pkg/admission/testing"
+	"k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/apis/core/helper"
+	"k8s.io/utils/ptr"
+)
+
+func TestAdmit(t *testing.T) {
+
+	plugin := admissiontesting.WithReinvocationTesting(t, newDisableServiceLinks())
+
+	tests := []struct {
+		description  string
+		requestedPod core.Pod
+		expectedPod  core.Pod
+		operation    admission.Operation
+	}{
+		{
+			description: "Create empty pod with default value of Spec.EnableServiceLinks",
+			requestedPod: core.Pod{
+				Spec: core.PodSpec{},
+			},
+			expectedPod: core.Pod{
+				Spec: core.PodSpec{EnableServiceLinks: ptr.To(false)},
+			},
+			operation: admission.Create,
+		},
+		{
+			description: "Create empty pod with Spec.EnableServiceLinks set to true",
+			requestedPod: core.Pod{
+				Spec: core.PodSpec{EnableServiceLinks: ptr.To(true)},
+			},
+			expectedPod: core.Pod{
+				Spec: core.PodSpec{EnableServiceLinks: ptr.To(false)},
+			},
+			operation: admission.Create,
+		},
+		{
+			description: "Update empty pod with Spec.EnableServiceLinks set to true",
+			requestedPod: core.Pod{
```

Comment:
An updated operation on a pod does not default to `Spec.EnableServiceLinks=false` but returns a `Spec.EnableServiceLinks=nil` instead.

I am assuming this is expected behavior where defaults are not injected in an updated operation.